src/main/java/com/direwolf20/buildinggadgets/common/packets/UpdateTemplatePacket.java

```diff
@@ -0,0 +1,211 @@
+            .build(new CacheLoader<Optional<UUID>, PacketAssembly>() {
+                @Override
+                public PacketAssembly load(@Nonnull Optional<UUID> key) {
+                    return new PacketAssembly();
+                }
+            });
+
+    public static void send(UUID id, @Nullable Template template, PacketTarget target) {
+        Optional.ofNullable(template)
+                .flatMap(TemplateIO::getCompressedBytes)
+                .map(b -> {
+                    int remainingBytes = b.length;
+                    int seqNumber = 0;
+
+                    while (remainingBytes > 0) {
+                        int size = Math.min(remainingBytes, SPLIT_SIZE);
+                        remainingBytes -= size;
+                        byte[] payload = new byte[size];
+                        System.arraycopy(b, seqNumber * SPLIT_SIZE, payload, 0, size);
+                        Packets.INSTANCE.send(target, new UpdateTemplatePacket(id, curSession.getAndIncrement(),
+                                payload, seqNumber++, remainingBytes > 0));
+                    }
+
+                    return null;
+                })
+                .orElseGet(() -> {
+                    Packets.INSTANCE.send(target,
+                            new UpdateTemplatePacket(id, curSession.getAndIncrement(), null, 0, false));
+                    return null;
+                });
+    }
+
+    @Nullable
+    private final byte[] payload;
+    private final boolean hasMore;
+    private final int session;
+    private final int seqNumber;
+
+    public UpdateTemplatePacket(UUID id, int curSession, @Nullable byte[] payload, int seqNumber, boolean hasMore) {
+        super(id);
+        this.session = curSession;
+        this.payload = payload;
+        this.seqNumber = seqNumber;
+        this.hasMore = hasMore;
+    }
+
+    public UpdateTemplatePacket(PacketBuffer buffer) {
+        super(buffer);
+        //ensure that it actually reads the size it wrote - the no-arg version reads the whole buffer!!!
+        session = buffer.readInt();
+
+        if (buffer.readBoolean()) {
+            payload = buffer.readByteArray(SPLIT_SIZE);
+            seqNumber = buffer.readVarInt();
+        } else {
+            payload = null;
+            seqNumber = - 1;
+        }
+
+        hasMore = buffer.readBoolean();
+    }
+
+    public void encode(PacketBuffer buffer) {
+        super.encode(buffer);
+
+        buffer.writeInt(session);
+        buffer.writeBoolean(payload != null);
+
+        if (payload != null)
+            buffer.writeByteArray(payload);
+    }
+
+    public void handle(Supplier<Context> ctx) {
+        ctx.get().enqueueWork(() -> {
+            try {
+                PacketAssembly assembly = assemblyByPlayerId.get(Optional.ofNullable(ctx.get().getSender()).map(Entity::getUniqueID));
+
+                Supplier<Supplier<World>> loadSafeWorld = ctx.get().getDirection().getReceptionSide().isClient() ?
```

Comment:
Can you not just use the server world here?

Reply:
As you can see by how it's used: No. I need to always get the cap.